core/src/main/scala/zio/logging/LogFormat.scala

```diff
@@ -82,4 +84,80 @@ object LogFormat {
       format(lineFormat(context, line), date, level, loggerName, maybeError)
     }
   }
+
+  object AssembledLogFormat {
+    type FormatterFunction = (StringBuilder, LogContext, String) => Any
```

Comment:
This should be a case class so you don't have to add syntax. In addition, the case class should store a function, but probably that function should be package private.

Reply:
Sounds good, I will refactor